demos/qml/tutorial_learningshallow.ipynb

```diff
@@ -0,0 +1,965 @@
+    "\n",
+    "    def step(adam_count, adam_mu, adam_nu, theta, val, min_energy):\n",
+    "        val, grad_circuit = cost(theta, index), grad(cost)(theta, index)\n",
+    "        updates, opt_state = optimizer.update(grad_circuit, tree_unflatten(opt_state_tree_def, [adam_count, adam_mu, adam_nu]))\n",
+    "        opt_state_list, _ = tree_flatten(opt_state)\n",
+    "        adam_count, adam_mu, adam_nu = opt_state_list\n",
+    "        theta = optax.apply_updates(theta, updates)\n",
+    "        min_energy = jnp.minimum(val, min_energy)\n",
+    "        return adam_count, adam_mu, adam_nu, theta, val, min_energy\n",
+    "\n",
+    "    for i in range(n_epochs):\n",
+    "        adam_count, adam_mu, adam_nu, theta, val, min_energy = step(adam_count, adam_mu, adam_nu, theta, val, min_energy)\n",
+    "\n",
+    "    return theta, val, min_energy\n",
+    "\n",
+    "## 2x wires for circuit sewing \n",
+    "dev = qml.device(\"lightning.qubit\", wires=n*2)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As a cost function, we perform state tomography after applying\n",
+    "$U^\\text{target}$ and our Ansatz $V_i$. Our aim is to bring the state on\n",
+    "qubit `i` back to the north pole of the Bloch sphere, and we specify our\n",
+    "cost function accordingly.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {
+    "collapsed": false,
+    "jupyter": {
+     "outputs_hidden": false
+    }
+   },
+   "outputs": [],
+   "source": [
+    "@qjit\n",
+    "@qml.qnode(dev)\n",
+    "def qnode_i(params, i):\n",
+    "    U_target(wires)\n",
+    "    V_i(params, wires)\n",
+    "    return [qml.expval(P(i)) for P in [X, Y, Z]]\n",
+    "    \n",
+    "def cost_i(params, i):\n",
+    "    x, y, z = qnode_i(params, i)\n",
+    "    return x**2 + y**2 + (1-z)**2"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can now run the optimization. We see that in that case it suffices to\n",
+    "use the random initial values from above for each optimization.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "metadata": {
+    "collapsed": false,
+    "jupyter": {
+     "outputs_hidden": false
+    }
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "final loss: 4.168309870147671e-05; min loss: 4.168309870147671e-05; after 0:00:01.881214\n",
+      "final loss: 4.426389047114445e-06; min loss: 4.426389047114445e-06; after 0:00:01.588220\n",
+      "final loss: 7.119522980852351e-06; min loss: 6.622691688632588e-06; after 0:00:01.690259\n",
+      "final loss: 3.795733855917371e-06; min loss: 3.4988574855745106e-06; after 0:00:01.623869\n"
+     ]
+    }
+   ],
+   "source": [
+    "params_i = []\n",
+    "for i in range(n):\n",
+    "    t0 = datetime.now()\n",
+    "    theta, val, min_energy = run_opt(cost_i, params, i)\n",
```

Comment:
I'm getting an error on this line:
```
TypeError: Value dtype('float64') with type <class 'numpy.dtypes.Float64DType'> is not a valid JAX type
```

Do you have the same problem or is that just me?

Reply:
I do not have the error. I think it could be caused by the commit version.

If line 700 does not have jnp conversion, the error will be triggered
https://github.com/PennyLaneAI/catalyst/blob/7a27666b7891450b2c76824ce9d4847049f894ce/demos/qml/tutorial_learningshallow.ipynb#L700

Reply:
Oh my bad, I didn't have your branch checked out. With the optax exclusion everything works 💯